### ISSUE TITLE
Live demo?

### ISSUE THREAD

**Issue by daviddoria** [Kitware/visualizer#28]
Is there an active install of Visualizer somewhere with some demo data that can be used to show someone the functionality without having to download and setup an accessible server myself?

**Comment by scottwittenburg** [comment]
I do not think we have a live one anywhere.  Or I should say I'm not aware of it if we do.  But it isn't too bad to run it yourself.  Just download any of the ParaView binaries, and run the command line for your OS as described here:

https://kitware.github.io/paraviewweb/docs/architecture.html#Simplicity

**Comment by daviddoria** [comment]
I would strongly recommend you add this. The link you've sent (running on localhost) still doesn't enable the "hey check this out: [link]" type of email that should be the thing you'd want people sending around.

XTK has a ton of demos: https://github.com/xtk/X

And there are plenty of paid services that do the same: (e.g. https://app.pointerra.io/tryit/)

You can imagine this would be an enormous potential deterrent when not-as-technical-as-you'd like people are trying to quickly compare the interface/features/performance of some of these tools before deciding which to use.

**Comment by jourdain** [comment]
Well if it does not require a server with GPU, definitely easy to do.
For that you can look at some [vtk.js examples](https://kitware.github.io/vtk-js/examples/ZipHttpReader.html).

But for Visualizer, it does require a ParaView process per user with a good network connection. 
Which works well on EC2 within a given region. But if someone from Europe/Asia/Australia try it out from a US server, he may think that the performance are not what he is looking for.

Just my 2 cents. Specially since it is super easy to demo locally.

**Comment by jourdain** [comment]
David, we are planning to setup such EC2 machine running full time and available for demos in the coming month. Hopefully this will allow what you just describe.

**Comment by llionevans** [comment]
Re: scottwittenburg's comment above

I've tried running the commands from the linked page
https://kitware.github.io/paraviewweb/docs/architecture.html#Simplicity

It looks very straightforward but I get the following error:
```
Traceback (most recent call last):
  File "./share/paraview-5.3/web/visualizer/server/pvw-visualizer.py", line 85, in <module>
    from paraview.web import wamp      as pv_wamp
ImportError: cannot import name wamp
```

Any ideas?

**Comment by jourdain** [comment]
Which version of ParaView?

**Comment by jourdain** [comment]
Is it on Linux with the distributed version of ParaView 5.3.

If so, this is a known bug of our distributed binary. 

To fix that you should use 5.2 or 5.4.

**Comment by llionevans** [comment]
I downloaded a fresh copy of 5.3 because the text on the documentation page shows that version.

```
$ cd ParaView-5.3.0-Qt4-OpenGL2-MPI-Linux-64bit
$ ./bin/pvpython                                                 \
    ./share/paraview-5.3/web/visualizer/server/pvw-visualizer.py  \
    --content ./share/paraview-5.3/web/visualizer/www/             \
    --data $PWD/share/paraview-5.3/data                         \
    --port 8080
==> Open your browser to http://localhost:8080/
```

Could the text be changed for clarity to avoid future confusion?

I've just just tried it on 5.2 and 5.4 and it works. Great to get a chance to interact with a working version rather than just seeing youtube videos, looks like an incredibly useful tool. Great work! Looking forward to putting it to use. Also, I'd be interested to see the EC2 deployment mentioned above when it's ready.

**Comment by jourdain** [comment]
Thanks!

Yes, we try to update it each time we release a new version of ParaView. So we should do it with 5.4.

**Comment by llionevans** [comment]
Just tried out Visualizer using ami-34f3f65e on EC2 (g2.2xlarge). It works well even though I'm logged into a US machine from the EU. Great work!

The link below says that you're working on an ami for v5.5. Any ideas of a timescale? Just trying to work out whether I should hold off for a bit before I work on integrating it into a website.
https://kitware.github.io/paraviewweb/docs/ami.html

**Comment by llionevans** [comment]
Even the lidar demo dataset is pretty smooth.

**Comment by jourdain** [comment]
Thanks for your feedback. We do have a repo with our web demo that is not yet public with all the manual steps (which could be scripted) needed to build paraview with EGL and configure paraviewweb to have Visualizer with other web app running on EC2 or on a local machine.

I don't have a time frame when such document could be released to the public or if we will make an AMI out of it.

If I didn't get back to you with a better timeframe by next week, try to ping me again.

Thanks!

Seb

**Comment by llionevans** [comment]
Apologies, this is absolutely not the place to post this but I wasn't quite sure what was the best way to get in touch. Just let me know if I should use the ParaView email list instead (or something else).

For me, a main motivation of using visualizer is to be able to share simulation results with a collaborator without them needing the hardware or software to open the data. Because of this we need to assume that the collaborator is also not familiar with ParaView.

What I'd like to do is preload a dataset in a saved state (e.g. using .pvsm). I've found documentation for visualizer regarding installation but not regarding usage. Could you please point me in the right direction?

Also, does visualizer accept all of the same input file types as ParaView? If not, is there a comprehensive list somewhere saying which types are compatible?

Again, sorry for using the GitHub issue tracker to post this question.

**Comment by jourdain** [comment]
No worries, maybe we can start a new thread/issue here https://github.com/Kitware/visualizer/issues

That way, those informations could be found by others as they are useful.

Thanks,

Seb

**Comment by llionevans** [comment]
Ok, created a new issue.
Cheers
